### PR TITLE
feat(module-tools): add transformLodash option to disable lodash feature

### PR DESCRIPTION
**Pull request by targeral**
### <samp>🤖 Generated by Copilot at dbd3e4d</samp>

This pull request adds a new feature to the `@modern-js/module-tools`, `@modern-js/module-tools-docs`, and `@modern-js/libuild-plugin-swc` packages that allows users to enable or disable the lodash feature in the SWC compiler. The feature reduces the bundle size of the library code that uses lodash by transforming the import statements. The pull request updates the documentation, configuration, types, utilities, and tests related to the feature in both English and Chinese. The pull request also fixes some typos, formatting, and linting issues in the code and documentation.

### <samp>🤖 Generated by Copilot at dbd3e4d</samp>

*  Add `transformLodash` option to `@modern-js/module-tools` package to enable or disable lodash feature ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-4a6bd7a4bd234caac0f3e9a23f71bd2eae6eb096a9a8b7a2616389c2ee6933e5R153),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-4a6bd7a4bd234caac0f3e9a23f71bd2eae6eb096a9a8b7a2616389c2ee6933e5R212),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-eefbcb5cbcf3594b5de4f212c4a6feb4d7215006a49f19d6848348afe6bfe005R185-R187),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-ba4c98ff890890d89ce89a86119cf794ab7e04b21796ec80d3b3fa373b23ef09R54),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-c396ea1ac3680532b4b931369f621014fab56d1b80aee366816b7398311e8027R120),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-3afd2b55e4c2b246951bbbc69619d3d9e54534d125b6dfead2de7f815d196e3fR130))
*  Add `lodash` option to `@modern-js/libuild-plugin-swc` package to implement lodash feature using SWC compiler ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-12171d128ab66638af212d4b9f6fe8fed2b0ac496d3b7f47f08091e2f42a1f4dR1-R6),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-55018f791bc2841854bf3ba8dd8ab06b9465440a5767942c9f76ef05dbfbd2f9R3-R6),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-93da8bbedfdc78cbff3b4b06bc863b3503216924b866f6a9e4e78b5b0aef4796L15-R25),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-93da8bbedfdc78cbff3b4b06bc863b3503216924b866f6a9e4e78b5b0aef4796L24-R131))
*  Add documentation for `transformLodash` option in both English and Chinese to `@modern-js/module-tools-docs` package ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-165dad5fb3ea2382f2c33d9a53baaf6dd56e07c937ee2062ce9c13b1f21f4f3dR1-R6),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-90a0f132c4140e37c110b40455aa3d8ad4d6a89ca7a8c8d19aa1b9a975b2d3baR1184-R1228),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-c586a6521b4022a3ce9cca9155cb7658a6d5808af1138b83eb5bcd5d11e48791R1184-R1226))
*  Fix error and formatting issues in English documentation of `buildConfig` option in `@modern-js/module-tools-docs` package ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-90a0f132c4140e37c110b40455aa3d8ad4d6a89ca7a8c8d19aa1b9a975b2d3baL647-R647),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-90a0f132c4140e37c110b40455aa3d8ad4d6a89ca7a8c8d19aa1b9a975b2d3baR1039-R1040),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-90a0f132c4140e37c110b40455aa3d8ad4d6a89ca7a8c8d19aa1b9a975b2d3baL1224))
*  Format import statements in `transform.ts` file in `@modern-js/libuild-plugin-swc` package ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-93da8bbedfdc78cbff3b4b06bc863b3503216924b866f6a9e4e78b5b0aef4796L4-R10))
*  Add integration test for `transformLodash` option in `@modern-js/module-tools` package using `transformLodash` fixture ([link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-8e6cbc3c34721bf715337098063e8d1367aed2c9099045a01743a0c758d2ccdeR1-R38),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-efe160366d92a54b267d5c7948a727f0efd604e5e88a133ff5fed3ae6bf8f928R1-R7),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-fbc31483a044729fea6ec9dbbd947682f19e400cec480aa81250a1d699524513R1-R5),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-d4287ae60351cc41766fea23ecb672426dc7107ddeadb2b9e2ae3e14d537e5a9R1-R5),[link](https://github.com/web-infra-dev/modern.js/pull/4468/files?diff=unified&w=0#diff-7e895a771c1fd86eb0ccbee79c1d139893eab13bb7405a00410a94f0eae3ffa4R1-R47))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/libuild/libuild-plugin-swc/src/transform.ts [hunk 1]
```diff
@@ -12,101 +18,117 @@ export interface SwcTransformOptions {
...
-              lodash: {
-                cwd: process.cwd(),
-                ids: ['lodash', 'lodash-es'],
+              isModule: 'unknown',
+              module,
+              extensions: {
+                pluginImport,
+                lodash: enableLodash
+                  ? {
+                      cwd: appDirectory,
```
Comment (targeral) [inline packages/libuild/libuild-plugin-swc/src/transform.ts:91]: Note the changes here.

File: .changeset/tame-vans-judge.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@modern-js/module-tools': minor
+---
+
+feat: add transformLodash option to enable or disable lodash feature
```
Comment (chenjiahan) [inline .changeset/tame-vans-judge.md:5]: please update the PR title~

File: packages/document/module-doc/docs/en/api/config/build-config.mdx [hunk 3]
```diff
@@ -644,7 +644,7 @@ For more information about JSX Transform, you can refer to the following links:
 
 - [React Blog - Introducing the New JSX Transform](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html).
 - [esbuild - JSX](https://esbuild.github.io/api/#jsx).
-  :::
+:::
```
Comment (chenjiahan) [inline packages/document/module-doc/docs/en/api/config/build-config.mdx:647]: Usually we need to add an blank line before `:::` to make prettier working correctly

File: packages/libuild/libuild-plugin-swc/src/transform.ts [hunk 4]
```diff
@@ -12,101 +18,117 @@ export interface SwcTransformOptions {
   useDefineForClassFields?: boolean;
   externalHelpers?: boolean;
   pluginImport?: ImportItem[];
+  lodash?: boolean;
```
Comment (chenjiahan) [inline packages/libuild/libuild-plugin-swc/src/transform.ts:21]: Can we use `tramsfornLodash` to make the option name consistent?

Comment (targeral) [reply packages/libuild/libuild-plugin-swc/src/transform.ts:21]: I think it's ok.

File: .changeset/tame-vans-judge.md (hunk 2 above)
Comment (targeral) [reply .changeset/tame-vans-judge.md:5]: Done

**Review by chenjiahan — DISMISSED** [review]

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**